### PR TITLE
audit 6.27,6.29

### LINKED ISSUE #155: 6.27 Transition CompleteWithdrawal Can Perform 0 Amount Transfers
In case `withdrawal_pending[initiator]` is an empty map or when no entries in this map meet the withdrawal criteria, the transfer will still be performed. The amount of the message will be 0 and the `initiator` will just burn the gas.

```
transition CompleteWithdrawal(initiator: ByStr20)
  IsPaused;
  IsProxy;
  withdraw_map_o <- withdrawal_pending[initiator];
  match withdraw_map_o with
  | Some withdraw_map =>
    withdraw_list = builtin to_list withdraw_map;
    (* To init tmp fileds *)
    withdrawal_zero = uint128_zero;
    avaiable_withdrawal := withdrawal_zero;
    c = Some {ByStr20} initiator;
    current_deleg := c;

    (* Calculate withdrawal that can be transferred *)
    forall withdraw_list CalculateTotalWithdrawal;
    withdraw_amt <- avaiable_withdrawal;
    e = { _eventname: "CompleteWithdrawal"; ssn_addr: initiator; initiator: withdraw_amt };
    event e;
    TransferFunds addfunds_tag withdraw_amt initiator
  | None =>
  end
end
```

### PR DESCRIPTION
**Pull request by renlulu**
Fixes #155 and #157

### CODE REVIEW

File: contracts/ssnlist.scilla [hunk 1]
```diff
@@ -1119,10 +1117,19 @@ transition CompleteWithdrawal(initiator: ByStr20)
...
-    e = { _eventname: "CompleteWithdrawal"; ssn_addr: initiator; initiator: withdraw_amt };
-    event e;
-    TransferFunds addfunds_tag withdraw_amt initiator
+    transfer = builtin eq withdraw_amt uint128_zero;
+    match transfer with
+    | True =>
+      e = { _eventname: "NoUnboundDelegate"; ssnaddr: initiator };
+      event e
+    | False =>
+      e = { _eventname: "CompleteWithdrawal"; ssnaddr: initiator; initiator: withdraw_amt };
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:1126]:
```suggestion
      e = { _eventname: "CompleteWithdrawal"; ssnaddr: initiator; amount: withdraw_amt };
```

File: contracts/ssnlist.scilla [hunk 2]
```diff
@@ -1119,10 +1117,19 @@ transition CompleteWithdrawal(initiator: ByStr20)
...
+      event e;
+      TransferFunds addfunds_tag withdraw_amt initiator
+    end
   | None =>
+    e = { _eventname: "NoDelegateToWithdral"; ssnaddr: initiator };
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:1131]:
```suggestion
    e = { _eventname: "NoPendingWithdrawal"; ssnaddr: initiator };
```

File: contracts/ssnlist.scilla [hunk 3]
```diff
@@ -1119,10 +1117,19 @@ transition CompleteWithdrawal(initiator: ByStr20)
     (* Calculate withdrawal that can be transferred *)
     forall withdraw_list CalculateTotalWithdrawal;
     withdraw_amt <- avaiable_withdrawal;
-    e = { _eventname: "CompleteWithdrawal"; ssn_addr: initiator; initiator: withdraw_amt };
-    event e;
-    TransferFunds addfunds_tag withdraw_amt initiator
+    transfer = builtin eq withdraw_amt uint128_zero;
+    match transfer with
+    | True =>
+      e = { _eventname: "NoUnboundDelegate"; ssnaddr: initiator };
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:1123]:
```suggestion
      e = { _eventname: "NoUnbondedStake"; ssnaddr: initiator };
```

**Review by AmritKumar — CHANGES_REQUESTED** [review]
Check the comments.

**Review by AmritKumar — APPROVED** [review]

**Review by bb111189 — APPROVED** [review]

**Pull request merged**